ui/conductor/src/traits/lighting/useLightingTrait.js

```diff
@@ -1,89 +1,133 @@
-import {closeResource, newActionTracker, newResourceValue} from '@/api/resource';
+import {newActionTracker, newResourceValue} from '@/api/resource';
 import {pullBrightness, updateBrightness} from '@/api/sc/traits/light';
-import {useErrorStore} from '@/components/ui-error/error';
-import {onMounted, onUnmounted, reactive, watch} from 'vue';
+import {toQueryObject, watchResource} from '@/util/traits.js';
+import {toValue} from '@/util/vue';
+import {computed, reactive} from 'vue';
 
 /**
  *
- * @param {Object} props
- * @param {string} props.name
- * @param {boolean} [props.paused]
+ * @template T
+ * @param {MaybeRefOrGetter<T>} query - The name of the device or a query object
+ * @param {MaybeRefOrGetter<boolean>} paused - Whether to pause the data stream
+ * @param {{
+ *  pullBrightness?: boolean,
+ *  describeBrightness?: boolean
+ * }} [options] - Options to control which API calls to make
  * @return {{
- *  lightValue: import('vue').UnwrapNestedRefs<
- *    ResourceValue<Brightness.AsObject, Brightness>
- *   >,
- *  updateValue: import('vue').UnwrapNestedRefs<
- *   ActionTracker<Brightness.AsObject>
- *  >,
- *  doUpdateBrightness: (req: number|Brightness.AsObject|UpdateBrightnessRequest.AsObject) => void
+ *  light: ResourceValue<Brightness.AsObject, Brightness>,
+ *  lightUpdate: ActionTracker<Brightness.AsObject>,
+ *  brightnessLevelNumber: import('vue').ComputedRef<number>,
+ *  brightnessLevelString: import('vue').ComputedRef<string>,
+ *  lightIcon: import('vue').ComputedRef<string>,
+ *  lightError: import('vue').ComputedRef<ResourceError|ActionError>,
+ *  doUpdateBrightness: (req: number|Brightness.AsObject|UpdateBrightnessRequest.AsObject) => void,
+ *  lightLoading: import('vue').ComputedRef<boolean>
  * }}
  */
-export default function(props) {
-  const errorStore = useErrorStore();
-
-  const lightValue = reactive(
+export default function(query, paused, options) {
+  const light = reactive(
       /** @type {ResourceValue<Brightness.AsObject, Brightness>} */
       newResourceValue());
 
-  const updateValue = reactive(
+  const lightUpdate = reactive(
       /** @type {ActionTracker<Brightness.AsObject>}  */
       newActionTracker());
 
+  // Make sure the query is an object
+  const queryObject = toQueryObject(query);
+
+  /**
+   * Create a list of API calls to make based on the options passed in
+   *
+   * @type {import('vue').ComputedRef<Array<(req: any) => any>>} apiCalls
+   */
+  const apiCalls = computed(() => [
+    // For light state and updates
+    ...(options?.pullBrightness !== false ? [(req) => {
+      pullBrightness(req, light);
+      return light;
+    }] : [])
+  ]);
 
+  // Utility function to call the API with the query and the resource
+  watchResource(
+      () => toValue(queryObject),
+      () => toValue(paused),
+      ...apiCalls.value
```

Comment:
There's no point in `apiCalls` being computed. Using it here in the function body of this composable negates the reactivity.

As `apiCalls` doesn't use any refs anyway you should just be able to either inline the array or convert `apiCalls` into a normal array.